From c8199c2f2962a5de77cc4425c76f0da7303fe649 Mon Sep 17 00:00:00 2001
From: Junxiao Bi <junxiao.bi@windriver.com>
Date: Sat, 31 Dec 2011 13:45:15 +0800
Subject: [PATCH 2/2] driver: w1: ds2438: fix memory leak when rmmod the driver

di is allocated using kzalloc() in ds2438_add_slave() and
should be freed in ds2438_remove_slave().

Signed-off-by: Junxiao Bi <junxiao.bi@windriver.com>
---
 drivers/w1/slaves/w1_ds2438.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/w1/slaves/w1_ds2438.c b/drivers/w1/slaves/w1_ds2438.c
index 8186cbc..4d0d4c9 100644
--- a/drivers/w1/slaves/w1_ds2438.c
+++ b/drivers/w1/slaves/w1_ds2438.c
@@ -574,6 +574,7 @@ static void ds2438_remove_slave(struct w1_slave *slave)
 					  &di->monitor_work);
 	destroy_workqueue(di->monitor_wqueue);
 	power_supply_unregister(&di->bat);
+	kfree(di);
 
 	/* Got the mutex again for the following steps in w1_slave_detach() */
 	mutex_lock(&slave->master->mutex);
-- 
1.7.0.4

